### PR TITLE
New vocab! text-commons

### PR DESCRIPTION
**Pull request by ptsefton**
Have added a new vocab "text-commons" - this is real work, not a test, we want to develop this vocabulary/

### CODE REVIEW

File: text-commons/vocabulary.csv [hunk 1]
```diff
@@ -0,0 +1,98 @@
+term,type,label,description,domain,range
+contributor,Property,"""Contributor""","""""",Person,Organization,Dataset
+translator,Property,"""Register""","""Specifies the type of register (any of the varieties of a language that a speaker uses in a particular social context [Merriam-Webster]) of the contents of a language resource
```
Comment (simleo) [inline text-commons/vocabulary.csv]: The label and description don't match the term here

File: text-commons/vocabulary.csv [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+""",Person,Organization,Dataset
```
Comment (mr-c) [inline text-commons/vocabulary.csv]: And this extra linebreak seems to break GitHub rendering..

**Comment by simleo** [comment]
In the csv file, the property rows have data for an extra column at the end that has no match in the header (the one filled with `Dataset`).

**Comment by simleo** [comment]
The csv file has quoting problems. The table is not read correctly both libreoffice calc and python's csv module. In particular, this section of the row for "responder":

```
"uh-huh", "amen", "you don't say"
```

gets split into multiple cells. A quoting method that works in this case (both in libreoffice writer and python) is the following:

* quote the whole field if it contains commas
* if the field contains double quotation marks, enter them twice

E.g.:

```
foobar,"equivalent of ""uh-huh"", ""amen"", ""you don't say"""
```

**Comment by dgarijo** [comment]
@ptsefton can you pls review the issues raised above?

**Comment by simleo** [comment]
The CSV still has problems, see https://github.com/Language-Research-Technology/ro-terms/pull/1. Additionally, there's a duplicate entry for `translator`

**Comment by ptsefton** [comment]
Sorry I'm not very good with these pull requests - I hope it's fixed now with the latest CSV

**Comment by simleo** [comment]
> Sorry I'm not very good with these pull requests - I hope it's fixed now with the latest CSV

Almost there :slightly_smiling_face:. Now quoted entries have an extra leading and trailing space, which also end up in `context.json` (which needs to be regenerated, since "contributor" and "Corpus" are gone and "register" has been added). Also, the description for `Bilingual Lexicon` appears truncated.

**Review by stain — APPROVED** [review]

**Comment by stain** [comment]
Sorry I merged it early just to get the names in, but we can still do the CSV updates on this branch.

**Comment by ptsefton** [comment]
Just checking on the status of this @stain & @simleo  - if I go to https://www.researchobject.org/ro-terms/text-commons/ the terms are not there - do I need to do anything?

**Comment by dgarijo** [comment]
@ptsefton,
your terms are at https://w3id.org/ro/terms/text-commons

**Comment by ptsefton** [comment]
I was expecting that they would resolve to a web page of some kind something like this: https://github.com/Language-Research-Technology/language-data-node-tools/blob/master/ontology.<https://github.com/Language-Research-Technology/language-data-node-tools/blob/master/ontology.md>





----------------------------

Dr Peter Sefton
Senior Technical Advisor, School of Languages and Culture
Mobile: 0404 096 932

________________________________

**Comment by dgarijo** [comment]
We don't have an html serialization for terms right now. But it's something we could add.

**Comment by ptsefton** [comment]
I think it's essential. One of the recomendations in the spec is to use things that resolve to an HTML page you can read.

Happy to help out with this.

----------------------------

Dr Peter Sefton
Senior Technical Advisor, School of Languages and Culture
Mobile: 0404 096 932

________________________________

**Comment by dgarijo** [comment]
Where would you want to redirect by default?
The table serialization in GitHub?https://github.com/ResearchObject/ro-terms/blob/master/example/vocabulary.csv

**Comment by ptsefton** [comment]
The CSV table is probably better than nothing, but I would lilke it to resolve to something equivalent to the Schema.org pages.

I have a markdown file that should work https://github.com/Language-Research-Technology/language-data-node-tools/blob/master/ontology.md
[https://opengraph.githubassets.com/92f41b1557fdba99554b7c575689b605b040f5ab16d91854a2d468bd59ad15b0/Language-Research-Technology/language-data-node-tools]<https://github.com/Language-Research-Technology/language-data-node-tools/blob/master/ontology.md>
language-data-node-tools/ontology.md at master · Language-Research-Technology/language-data-node-tools<https://github.com/Language-Research-Technology/language-data-node-tools/blob/master/ontology.md>
Miscellaneous Node/Javascrip utilities for preparing Arkisto repositories / RO-Crates with language data - language-data-node-tools/ontology.md at master · Language-Research-Technology/language-da...
github.com


Is there a way to send that with the PR?

This was generated from:

An Excel spresdsheeet, turned into a Crate using rocxl (from ro-crate-html-js), then into a markdown file using the rocschema script (this can also generate the RO-Terms CSV)

(the markdown file is out of sync with the terms I sent before, still working on this)

----------------------------

Dr Peter Sefton
Senior Technical Advisor, School of Languages and Culture
Mobile: 0404 096 932

________________________________

**Comment by dgarijo** [comment]
Peter,
we can have many different HTML serializations. In fact, I have tools to generate documentation pages such as the one in https://w3id.org/example/. The problem is that we need **the same** structure in all directories to ensure the redirection to the HTML is done consistently. The easiest way is redirecting to the github HTML. Or we can have a small script to generate the HTML from the context.

**Pull request merged**